source/slang/slang-emit-c-like.cpp

```diff
@@ -969,10 +969,11 @@ bool CLikeSourceEmitter::shouldFoldInstIntoUseSites(IRInst* inst)
         type = arrayType->getElementType();
     }
 
-    // Don't allow temporaries of pointer types to be created.
+    // Don't allow temporaries of pointer types to be created,
+    // if target langauge doesn't support pointers.
     if(as<IRPtrTypeBase>(type))
     {
-        return true;
+        return !supportPtrTypes();
```

Comment:
`supportPtrTypes()` is maybe a bit ambiguous. I might argue for either `doesTargetSupportPtrTypes()` or `arePtrTypesSupported()` since both of those make it clear just from the name that they return a `bool` based on whether or not pointer types are supported.

Reply:
Done.